### PR TITLE
refactor: Remove dependency on Boost::program_options from Algorithms/Writers/Readers

### PR DESCRIPTION
**Pull request by benjaminhuth**
This aims to be the first step into making the binaries optional to build by removing the dependencies on `Boost::program_options` from the Algorithms/Writers/Readers.

The second step would be to also unwind this from the Detectors, but this is a bit more complicated, and thus I would do this step by step.

A long term goal would be that only the Analysis Apps depend on Boost program options, so that just building the Examples does not need any Boost components that require binaries (since we also use `std::filesystem` now).

### CODE REVIEW

**Comment by paulgessinger** [comment]
@benjaminhuth do you want feedback now already or not yet?

**Comment by benjaminhuth** [comment]
If you have some general feedback, go ahead. I think feedback on implementation details should wait until this compiles^^

File: Examples/Detectors/ContextualDetector/include/ActsExamples/ContextualDetector/AlignedDetector.hpp [hunk 1]
```diff
@@ -20,7 +20,7 @@ namespace ActsExamples::Contextual {
 class InternallyAlignedDetectorElement;
 class InternalAlignmentDecorator;
 
-class AlignedDetector : public ActsExamples::IBaseDetector {
+class AlignedDetector {
```
[the remark this answers is not in the document]
Comment (benjaminhuth) [reply Examples/Detectors/ContextualDetector/include/ActsExamples/ContextualDetector/AlignedDetector.hpp]: Hmm I guess that are leftovers of my attempts to remove the dependencies also from the detectors, I forgot to clean those up again...

File: Examples/Detectors/DD4hepDetector/include/ActsExamples/DD4hepDetector/DD4hepDetector.hpp [hunk 2]
```diff
@@ -18,7 +17,7 @@
 namespace ActsExamples {
 namespace DD4hep {
 
-struct DD4hepDetector : public IBaseDetector {
+struct DD4hepDetector {
```
[the remark this answers is not in the document]
Comment (benjaminhuth) [reply Examples/Detectors/DD4hepDetector/include/ActsExamples/DD4hepDetector/DD4hepDetector.hpp]: same as above

File: Examples/CMakeLists.txt [hunk 3]
```diff
@@ -2,6 +2,6 @@ add_subdirectory(Algorithms)
 add_subdirectory(Detectors)
 add_subdirectory(Framework)
 add_subdirectory(Io)
-add_subdirectory(Run)
+add_subdirectory_if(Run ACTS_BUILD_EXAMPLES_BINARIES)
```
[the remark this answers is not in the document]
Comment (benjaminhuth) [reply Examples/CMakeLists.txt:5]: yeah also some not cleaned up left-overs...
As mentioned in a first attemt I tried to do all at once, but now that should be removed.

File: Examples/Run/Common/include/ActsExamples/Detector/IBaseDetector.hpp [hunk 4]
```diff
@@ -8,7 +8,7 @@
 
 #pragma once
 
-#include "ActsExamples/Utilities/OptionsFwd.hpp"
+#include "ActsExamples/Options/OptionsFwd.hpp"
```
[the remark this answers is not in the document]
Comment (benjaminhuth) [reply Examples/Run/Common/include/ActsExamples/Detector/IBaseDetector.hpp]: I would leave it there until we fully remove this

**Comment by benjaminhuth** [comment]
I think I've cleaned this up, if the CI goes through could you have a look again @paulgessinger ?

**Comment by benjaminhuth** [comment]
@paulgessinger the CI is running through now :)

**Pull request merged**